unit_test/batched/Test_Batched_SerialGemm.hpp

```diff
@@ -66,6 +66,96 @@ namespace Test {
       Kokkos::Profiling::popRegion();
     }
   };
+
+template<typename DeviceType,
+        typename ViewType,
+        typename ScalarType,
+        typename ParamTagType, 
+        typename AlgoTagType>
+  void impl_test_batched_gemm_half(const int N, const int matAdim1, const int matAdim2, const int matBdim1, const int matBdim2,
```

Comment:
why is this called impl_test_batched_half when nothing in this function actually refers to whether anything is half precision?

Reply:
actually it looks like there is FP16_EPSILON? Why not just make a conditional there if the rest of the code is the same as the "non-half" version?

Reply:
> why is this called impl_test_batched_half when nothing in this function actually refers to whether anything is half precision?

same as https://github.com/kokkos/kokkos-kernels/pull/794#discussion_r496779387.



> actually it looks like there is FP16_EPSILON? Why not just make a conditional there if the rest of the code is the same as the "non-half" version?

we could use the same approach with respect to always generating random numbers in single precision and checking results on the host in single precision. would this be an acceptable approach for complex numbers?